service-base/src/main/java/org/eclipse/hono/service/management/credentials/GenericCredential.java

```diff
@@ -46,12 +56,28 @@ public String getType() {
         return this.secrets;
     }
 
-    public void setSecrets(final List<GenericSecret> secrets) {
+    /**
+     * Sets the list of secrets to use for authenticating a device to protocol adapters.
+     * <p>
+     * The list cannot be empty and each secret is scoped to its validity period.
```

Comment:
obviously, it can ... maybe you want to add a check and throw an IllegalArgumentException in that case?

Reply:
ok 👍